libgnucash/engine/gnc-rational-rounding.hpp

```diff
@@ -72,8 +74,23 @@ round(T num, T den, T rem, RT2T<RoundType::floor>)
 //              << ", and rem " << rem << ".\n";
     if (rem == 0)
         return num;
-    if (num < 0)
-        return num + 1;
+    // floor num==0 that is the quotient of two numbers with opposite signs
+    if (num < 0 || (num == 0 && (SIGN_NUM(rem) * SIGN_NUM(den) < 0)))
+        return num - 1;
```

Comment:
I also fixed a mistake in flooring and the corresponding test that was rounding up negatives.